scripts/esbuild/internal-platform-hydrate.ts

```diff
@@ -0,0 +1,89 @@
+import type { BuildOptions as ESBuildOptions } from 'esbuild';
+import fs from 'fs-extra';
+import { join } from 'path';
+
+import { getBanner } from '../utils/banner';
+import { bundleDts } from '../utils/bundle-dts';
+import { BuildOptions } from '../utils/options';
+import { writePkgJson } from '../utils/write-pkg-json';
+import { externalAlias, getBaseEsbuildOptions, getEsbuildAliases, getEsbuildExternalModules } from './util';
+
+/**
+ * Create objects containing ESbuild options for the two bundles comprising
+ * the hydrate platform. This also performs relevant side-effects, like
+ * clearing out a directory and writing a `package.json` script to disk.
+ *
+ * @param opts build options
+ * @returns an array of ESBuild option objects
+ */
+export async function getInternalPlatformHydrateBundles(opts: BuildOptions): Promise<ESBuildOptions[]> {
+  const inputHydrateDir = join(opts.buildDir, 'hydrate');
+  const hydrateSrcDir = join(opts.srcDir, 'hydrate');
+  const outputInternalHydrateDir = join(opts.output.internalDir, 'hydrate');
+
+  await fs.emptyDir(outputInternalHydrateDir);
+
+  // write @stencil/core/internal/hydrate/package.json
+  writePkgJson(opts, outputInternalHydrateDir, {
+    name: '@stencil/core/internal/hydrate',
+    description:
+      'Stencil internal hydrate platform to be imported by the Stencil Compiler. Breaking changes can and will happen at any time.',
+    main: 'index.js',
+  });
+
+  await createHydrateRunnerDtsBundle(opts, inputHydrateDir, outputInternalHydrateDir);
+
+  const hydratePlatformInput = join(hydrateSrcDir, 'platform', 'index.js');
+
+  const external = [...getEsbuildExternalModules(opts, outputInternalHydrateDir), '@stencil/core/mock-doc'];
+
+  const internalHydrateAliases = getEsbuildAliases();
+  internalHydrateAliases['@platform'] = hydratePlatformInput;
+
+  const internalHydratePlatformBundle: ESBuildOptions = {
+    ...getBaseEsbuildOptions(),
+    entryPoints: [hydratePlatformInput],
+    format: 'esm',
+    platform: 'node',
+    outfile: join(outputInternalHydrateDir, 'index.js'),
```

Comment:
The output file here looks like it's 50% larger than the rollup-built equivalent. It looks like it's mostly a minification thing - the rollup output has [terser-based output from the prettyMinifyPlugin](https://github.com/ionic-team/stencil/blob/eca261e500082632b38e900b98dbf58de0af53d2/scripts/bundles/internal-platform-hydrate.ts#L80), whereas this doesn't. Does that make sense? Or is there something else I'm missing that could be contributing to this size jump?

Reply:
yeah that's my understand of the scale of the differences, it comes mainly from `const -> var`, some variable renaming, and things like that.

You can look at a diff for this file of the Esbuild output vs a modified version of the Rollup output where I've disabled the `prettyMinifyPlugin` here: https://github.com/alicewriteswrongs/stencil-internal-esbuild-diff/commit/d5de5b8853c6c0eff0bc4a2f5a410ff203439232#diff-c24c5d15551298c70c449fb27374fd9da9190c20caf5956db19b865a9d044f7a